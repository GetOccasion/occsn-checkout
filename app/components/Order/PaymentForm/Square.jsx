import React, { PureComponent } from 'react';

import occsn from '../../../libs/Occasion';

import SquareAPI from 'square';

import { Col, FormGroup, Input, Label, Row } from 'reactstrap';

import PaymentServiceProvider from './PaymentServiceProvider';

export default class Square extends PaymentServiceProvider {
  // Initializes the iFrame using the global SqPaymentForm library imported as SquareAPI
  initializeForm() {
    this.sqPaymentForm = new SquareAPI({

      // Initialize the payment form elements
      applicationId: global.OCCSN.square_key,
      inputClass: 'form-control-square',
      inputStyles: [{
        padding: '0.375em 0.75em',
        fontSize: '1em',
        lineHeight: 1.5,
        color: '#495057',
        backgroundColor: '#fff',
      }],

      // Initialize the credit card placeholders
      cardNumber: {
        elementId: 'sq-card-number',
        placeholder: '•••• •••• •••• ••••'
      },
      cvv: {
        elementId: 'sq-cvv',
        placeholder: 'CVV'
      },
      expirationDate: {
        elementId: 'sq-expiration-date',
        placeholder: 'MM/YY'
      },
      postalCode: {
        elementId: 'sq-postal-code',
        placeholder: '#####'
      },

      // SqPaymentForm callback functions
      callbacks: {
        cardNonceResponseReceived: (errors, nonce) => {
          if (errors) {
          } else {
            this.paymentMethodDeferred.resolve(occsn.CreditCard.build({ id: nonce }));
          }
        },
        unsupportedBrowserDetected: () => {},
        inputEventReceived: (inputEvent) => {
          switch (inputEvent.eventType) {
            case 'focusClassAdded':
              /* HANDLE AS DESIRED */
              break;
            case 'focusClassRemoved':
              /* HANDLE AS DESIRED */
              break;
            case 'errorClassAdded':
              /* HANDLE AS DESIRED */
              break;
            case 'errorClassRemoved':
              /* HANDLE AS DESIRED */
              break;
            case 'cardBrandChanged':
              /* HANDLE AS DESIRED */
              break;
            case 'postalCodeChanged':
              /* HANDLE AS DESIRED */
              break;
          }
        },
        paymentFormLoaded: () => {}
      }
    });

    this.sqPaymentForm.build();
  }

  // Triggers cardNonceResponseReceived event
  tokenizePaymentMethodData() {
    this.sqPaymentForm.requestCardNonce();
  }

  render() {
    return <section>
      <div id="sq-ccbox">
        <FormGroup>
          <Label>Card Number</Label>
          <div id="sq-card-number"></div>
        </FormGroup>

        <FormGroup>
          <Row>
            <Col xs="6">
              <Label>Expiration Date</Label>
              <div id="sq-expiration-date"></div>
            </Col>
            <Col xs="3">
              <Label>CVV</Label>
              <div id="sq-cvv"></div>
            </Col>
          </Row>
        </FormGroup>

        <FormGroup>
          <Label>Postal Code</Label>
          <div id="sq-postal-code"></div>
        </FormGroup>

        <input type="hidden" id="card-nonce" name="nonce" />
      </div>
    </section>;
  }
}
