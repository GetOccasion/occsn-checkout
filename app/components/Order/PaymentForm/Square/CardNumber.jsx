import React from 'react'

import { FormGroup, Label, FormFeedback } from 'reactstrap'

import { ErrorsFor } from 'mitragyna'

export default class CardNumber extends React.Component {
  shouldComponentUpdate() {
    return false
  }

  render() {
    return <div id="sq-card-number" />
  }
}
