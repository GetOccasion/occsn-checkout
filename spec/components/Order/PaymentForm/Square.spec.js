import React from 'react';

import Square from 'app/components/Order/PaymentForm/Square';

import SquareAPI from 'square';

import occsn from 'app/libs/Occasion';

describe('Order', () => {
  describe('PaymentForm', () => {
    describe('Square', () => {
      let wrapper;

      beforeEach(() => {
        wrapper = mount(<Square></Square>);
      });

      it('renders fields', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('calls SquareAPI constructor', () => {
        expect(SquareAPI.mock.calls.length).toBe(1);
      });

      describe('tokenizePaymentMethodData', () => {
        beforeEach(() => {
          wrapper.instance().tokenizePaymentMethodData();
        });

        it('calls SquareAPI.requestCardNonce', () => {
          expect(SquareAPI.getSqPaymentForm().requestCardNonce.mock.calls.length).toBe(1);
        })
      });
    });
  });
});
