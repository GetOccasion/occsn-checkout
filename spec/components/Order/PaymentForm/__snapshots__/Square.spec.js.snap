// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order PaymentForm Square renders fields 1`] = `
<Square>
  <section>
    <div
      id="sq-ccbox"
    >
      <FormGroup
        tag="div"
      >
        <div
          className="form-group"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
            >
              Card Number
            </label>
          </Label>
          <div
            id="sq-card-number"
          />
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="form-group"
        >
          <Row
            tag="div"
          >
            <div
              className="row"
            >
              <Col
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="6"
              >
                <div
                  className="col-6"
                >
                  <Label
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <label
                      className=""
                    >
                      Expiration Date
                    </label>
                  </Label>
                  <div
                    id="sq-expiration-date"
                  />
                </div>
              </Col>
              <Col
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
                xs="3"
              >
                <div
                  className="col-3"
                >
                  <Label
                    tag="label"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <label
                      className=""
                    >
                      CVV
                    </label>
                  </Label>
                  <div
                    id="sq-cvv"
                  />
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </FormGroup>
      <FormGroup
        tag="div"
      >
        <div
          className="form-group"
        >
          <Label
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              className=""
            >
              Postal Code
            </label>
          </Label>
          <div
            id="sq-postal-code"
          />
        </div>
      </FormGroup>
      <input
        id="card-nonce"
        name="nonce"
        type="hidden"
      />
    </div>
  </section>
</Square>
`;
